File: src/main/java/com/dv/telegram/command/ListAdmins.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;
import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.List;

public class ListAdmins extends BasicBotCommand {

    @Override
    public String getName() {
        return ListAdmins.class.getSimpleName();
    }

    @Override
    public String getDescription(WikiBot bot) {
        return String.format(
            "`%s %s` — вывести список администраторов бота, которые имеют право запускать специальные команды из списка в `%s %s`.%n%nКоманда `%s` всегда доступна всем пользователям.",
            bot.getBotName(),
            getCommandText(),
            bot.getBotName(),
            bot.getSpecialCommands().getListCommands().getCommandText(),
            bot.getSpecialCommands().getStartCommand().getCommandText()
        );
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/listAdmins";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        List<String> adminLines = new ArrayList<>();

        for (String botAdmin : bot.getSpecialCommands().getBotAdmins()) {
            adminLines.add(String.format(
                "— %s",
                getSettingValueForMarkdown( // escape _ in user names like "dmitry_weirdo"
                    BotCommandUtils.getClickableUserName(botAdmin)
                )
            ));
        }

        return StringUtils.join(adminLines, "\n");
    }
}

File: src/main/kotlin/com/dv/telegram/command/ListAdmins.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class ListAdmins : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) = """
        `${bot.botName} $commandText` — вывести список администраторов бота, которые имеют право запускать специальные команды из списка в `${bot.botName} ${bot.specialCommands.listCommands.commandText}`.
        
        Команда `${bot.specialCommands.startCommand.commandText}` всегда доступна всем пользователям.
        """.trimIndent()

    override fun useMarkdownInResponse() = true

    override val defaultCommandName = "/listAdmins"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val adminLines = mutableListOf<String>()

        for (botAdmin in bot.specialCommands.botAdmins) {
            adminLines.add(
                "— ${ getSettingValueForMarkdown(
                    BotCommandUtils.getClickableUserName(botAdmin)
                ) }" // escape _ in user names like "dmitry_weirdo"
            )
        }

        return adminLines.joinToString("\n")
    }
}
